import "goby/common/protobuf/option_extensions.proto";
import "goby/acomms/protobuf/driver_base.proto";
import "goby/acomms/protobuf/modem_message.proto";
import "goby/common/protobuf/app_base_config.proto";

package goby.test.protobuf;

message MMDriverTest2Config
{
    optional AppBaseConfig app = 1;

    required goby.acomms.protobuf.DriverConfig mm1_cfg = 10
        [(goby.field).description = "Configure the transmitting modem"];
    required goby.acomms.protobuf.DriverConfig mm2_cfg = 11
        [(goby.field).description = "Configure the receiving modem"];

    repeated goby.acomms.protobuf.ModemTransmission transmission = 20
        [(goby.field).description = "Sequence of transmissions to perform"];

    optional int32 repeat = 21 [
        (goby.field).description = "Repeat `transmission` this number of times",
        default = 1
    ];
}
